docs/Deploying/TUF-Update-Guide.md

```diff
@@ -0,0 +1,188 @@
+# TUF Update Guide
+
+This document is a walkthrough guide for how a Fleet member can become a publisher of updates for [Fleet's TUF service](tuf.fleetctl.com).
+The roles needed to push new updates are `targets`, `snapshot` and `timestamp`.
+
+## Security
+
+- TUF keys for `targets`, `snapshot` and `timestamp` should be stored on a USB stick.
+- The keys are stored encrypted with a passphrase stored in 1Password.
+
+## Sync Fleet's TUF repository
+
+For simplicity we sync the repository to the USB stick.
+```sh
+cd /Volumes/FLEET-TUF/repository
+
+cd /Volumes/FLEET-TUF
+mkdir -p repository
+mkdir -p keys
+mkdir -p staged
+
+aws s3 sync s3://fleet-tuf-repo ./repository
+```
+
+## Generate targets+snapshot+timestamp keys
+
+```sh
+tuf gen-key targets
+Enter targets keys passphrase:
+Repeat targets keys passphrase:
+Generated targets key with ID ae943cb8be8a849b37c66ed46bdd7e905ba3118c0c051a6ee3cd30625855a076
+```
+```sh
+tuf gen-key snapshot
+Enter snapshot keys passphrase:
+Repeat snapshot keys passphrase:
+Generated snapshot key with ID 1a4d9beb826d1ff4e036d757cfcd6e36d0f041e58d25f99ef3a20ae3f8dd71e3
+```
+```sh
+tuf gen-key timestamp
+Enter timestamp keys passphrase:
+Repeat timestamp keys passphrase:
+Generated timestamp key with ID d940df08b59b12c30f95622a05cc40164b78a11dd7d408395ee4f79773331b30
+```
+
+Share `staged/root.json` with Fleet member with the `root` role, who will sign with its root key and push to the repository.
+
+## Root role signs the `staged/root.json`
+
+Essentially the following commands are executed to sign the new keys:
+- `tuf sign`
+- `tuf snapshot`
+- `tuf timestamp`
+- `tuf commit`
+
+## Pushing new updates
+
+Following are tested steps to push new targets.
+
+### 1. Backup current TUF repository
+
+Just in case we break the remote TUF directory, let's do a local backup:
+```sh
+mkdir ~/tuf.fleetctl.com/backup
+aws s3 sync s3://fleet-tuf-repo ~/tuf.fleetctl.com/backup
+```
+
+### 2. Make sure the local repository is up-to-date
+
+```sh
+aws s3 sync s3://fleet-tuf-repo ./repository
+```
+
+### 3. Setup Orbit in Linux, Windows, macOS
+
+Install Orbit with the (to be updated) channels in the three supported OSs.
```

Comment:
Does this mean all three channels `orbit-channel`, `osqueryd-channel`, and `desktop-channel`?

Reply:
Correct. User has to install Orbit with the targets+channels being updated.

E.g. if I'm pushing a new version of osqueryd to `edge`, then I will generate and install a package with `--osqueryd-channel=edge` .